### ISSUE TITLE
Unable to deploy NEP5 sample smart contract

### ISSUE THREAD

**Issue by devhawk** [neo-project/neo-gui#274]
I'm following the [NGD Quickstart](https://github.com/neo-ngd/NEO-Tutorial/blob/master/neo_docs_SmartContract_QuickStart/SmartContract_quickstart_en.md#developing-and-deploying-nep-5-contract). I've successfully set up a private chain, transferred the genesis NEO to a standard account, waited until that account has several thousand GAS and then claimed it. I copy/pasted the [NEP-5 code](https://github.com/neo-ngd/NEO-Tutorial/blob/master/neo_docs_SmartContract_QuickStart/SmartContract_quickstart_en.md#editing-nep-5-code) into a new [C# NeoContract project](https://github.com/neo-ngd/NEO-Tutorial/blob/master/neo_docs_SmartContract_QuickStart/SmartContract_quickstart_en.md#creating-a-neo-contract-project), compiled it, then ran neon manually so I could specify the `--compatible` flag as per the quickstart. So far, so good.

Next, I follow the [Deploying contract](https://github.com/neo-ngd/NEO-Tutorial/blob/master/neo_docs_SmartContract_QuickStart/SmartContract_quickstart_en.md#deploying-contract) section of the quickstart, I select Advanced-Deploy Contract, fill in the information section of the Deploy Contract dialog, input 0710 for parameter list and 05 for return type, press the load button and select the .avm file, copy the script hash and select "Need Storage". Pressing "Deploy" brings up the Invoke Contract dialog, on the "Custom" tag. 

It's not really clear what to do next at this point. The Quickstart instructs the user to click `trial run`, "Double check" (what the user is supposed to double check is not clear) and then click `Invoke`. In the 2.10 version of NEO-GUI, there is an enabled `Test` button and a disabled `Invoke` button. Clicking `Test` pops a dialog reading "Execution terminated in fault state". The results pain of the Invoke Contract dialog reads

```
VM State: FAULT, BREAK
Gas Consumed: 500
Evaluation Stack: []
```

The error message isn't really clear here, but it appears the smart contract is failing to deploy. I've tried much simpler smart contracts, including ones that simply return a boolean value that is passed in as well as one that simply writes hello world to storage w/o taking or returning any parameters. I am not able to deploy any of them. I also tried using the NEO-CLI, and got similar results. 

Is there a more recent walkthru showing how to deploy a smart contract to a private NEO chain? Are there any troubleshooting guides for diagnosing smart contract deployment issues? Is this possibly an issue in NEO itself, given that both NEO-CLI and NEO-GUI failed to deploy?

**Comment by erikzhang** [comment]
Duplicate of #273

**Comment by igormcoelho** [comment]
I'm confident this may be caused by not attaching enough `Network Fee` for your transaction (it is probably bigger than 1024 bytes). Please try a simple NEP-5 on https://neocompiler.io, it will fail if you not attach fee (then you can try again attaching fee and it works). For testing, just add 1 GAS, it should be enough.
P.S.: sorry but I don't know neo-gui interface very well... there should be some easy way to do this.

**Comment by devhawk** [comment]
@igormcoelho I don't think it's an issue with fee since I had the issue w/ a hello world smart contract. NEP5.avm is over 2k , but the hello world .avm I tried was less than 100 bytes. And if there's a way to add GAS fee to a contract deployment in NEO CLI or NEO GUI, I don't see it

**Comment by devhawk** [comment]
@erikzhang #273 suggests this is a NEO GUI issue, but I'm seeing the problem in NEO CLI as well.

**Comment by Lichen9618** [comment]
@devhawk 
1. In "Invoke Contract dialog", your steps are correct.

2. About the contract cannot run successfully.
You can record the "Script Hash" from "deploy contract dialog". 
And search it in "Function Invoke" of "Invoke Contract dialog"
If the information can be checked as you deploy, then contract is deployed on chain.
If not, it seems other the problem.  We need further information to check.

**Comment by devhawk** [comment]
Good to know my steps are correct. Some of the docs are out of date, so it's sometimes hard to tell if I'm doing the right thing. Thanks @Lichen9618.

I built a very simple "negator" smart contract that simply NOTs and returns a boolean parameter. .avm file is only 31 bytes so it shouldn't hit the size limit @igormcoelho mentioned. I deployed it following my original steps and get the same "Execution terminated in fault state" dialog when I click Test. I copied the script hash from the Deploy Contract dialog before deployment. I switched over to the Function tab, pasted the script hash into the ScriptHash text box and pressed the search button (i.e. the button directly to the right of the ScriptHash text box with a picture of a magnifying glass). Nothing happened. So it appears the contract is not getting deployed. 

I've included my very simple negator smart contract below + attached a zip with the compiled .avm file as well as the .abi.json file that is also generated during compilation. Let me know what other further information is needed to troubleshoot this issue. 

``` csharp
using Neo.SmartContract.Framework;

namespace NegatorContract
{
    public class Contract1 : SmartContract
    {
        public static bool Main(bool value)
        {
            return !value;
        }
    }
}
```
[NegatorContract.zip](https://github.com/neo-project/neo-gui/files/3008504/NegatorContract.zip)

**Comment by igormcoelho** [comment]
@devhawk I created a tutorial to help you (using your proposed problem). You can test on neocompiler.io platform until this issue is fixed on neo-gui: https://medium.com/neoresearch/neo-compiler-eco-example-3-negation-contract-c-9096f272e03d

**Comment by Lichen9618** [comment]
@devhawk 
The code you provided can be deployed on my PC and neo-compiler successfully.
So we can confirm the contract code is correct.

Can you show me the Hex code auto generated after deploy contract, In "costume" tab?
This is transaction Hex code.
I can compare it with my own. To find the issue.

**Comment by shargon** [comment]
The contract works as expected in our tests (local environment without gas neither chain):

![image](https://user-images.githubusercontent.com/3167973/55059282-d6b01780-506e-11e9-97a5-70dd406ea694.png)

Did you try it in `release` mode?

**Comment by Edelbi1204** [comment]
i am having the same issue, i have tried and follow all guides with no hop ...
the same contract has been complied and tested successfully on neocompiler.io but still getting the same error when trying to deploy the contract on my private development chain using neo-gui
shall we expect a clear clarification for the cause of this issue ..!!!

**Comment by devhawk** [comment]
Sorry, for my delayed response. I got pulled away from this for a bit unavoidably. 

@igormcoelho I will take a look at your article and the neocompiler.io website. Thanks!

@shargon When you ask if I tried it in release mode, it's not clear to me what you are referring to. Are you asking if I'm running release builds? Also, what tool is that in the screenshot you posted?

@Lichen9618, here is the hex from the custom tab of the invoke contract:

```
0a6974206e656761746573116861727279406465766861776b2e6e65740548617272790131076e656761746f72005101011f52c56b6c766b00527ac4616a00c3009c6a51527ac46203006a51c3616c756668134e656f2e436f6e74726163742e437265617465
```

**Comment by devhawk** [comment]
Based on the thread for #273 and @shargon's question about release, I tried running neo-gui from source. I got a completely different stream of hex values in the Invoke Contract dialog - though I now realize that might be because I didn't use exactly the same contract metadata values. But more importantly, it worked - I was able to go to the Function tab of the Invoke Contract dialog, search for the deployed contract by ScriptHash and invoke it. 

I'm going to do more digging here - I wrote above that I had the same issue w/ neo-cli. But this now appears to be more of an issue with the latest released version of neo-gui (Note, the version of neo-gui available from the release page is 2.9.0 and is 8 months old). But at least I'm unblocked now.

New hex stream:
```
076e656761746f720d486172727940666f6f2e6261720548617272790131074e656761746f72005101011f52c56b6c766b00527ac4616a00c3009c6a51527ac46203006a51c3616c756668134e656f2e436f6e74726163742e437265617465
```

**Comment by devhawk** [comment]
FYI, I was able to take the original stream of hex info that didn't work with the currently shipping neo-gui from my earlier comment and copy/paste it into the custom tab of the Invoke Contract dialog, deploy it to my private chain and invoke it. (I reset the chain first so that I could be sure I wasn't interacting with an version of the contract I had deployed previously). So whatever is happening here appears to be an issue w/ neo-gui 2.9.0.

**Comment by jmseigneur** [comment]
Same problem here after setting up a privatenet following https://docs.neo.org/en-us/network/private-chain/private-chain2.html I'm able to deploy a simple contract with NEO-GUI but not a more complex contract such as the one of the NEP5 example

**Comment by hal0x2328** [comment]
This was fixed here: https://github.com/neo-project/neo-gui/commit/4798a696f3f952af2ca69c605a33e28757e9a5f2

A new release is all that is needed now.

**Comment by Ashuaidehao** [comment]
@lock9 this issue was resolved and can be closed.

**Comment by shargon** [comment]
Closed because is solved
